src/main/java/org/opensearch/flowframework/workflow/CreateIngestPipelineStep.java

```diff
@@ -0,0 +1,199 @@
+                        inputFieldName = (String) content.get(INPUT_FIELD);
+                        break;
+                    case OUTPUT_FIELD:
+                        outputFieldName = (String) content.get(OUTPUT_FIELD);
+                        break;
+                }
+            }
+
+            // Determmine if fields have been populated, else iterate over remaining workflow data
+            if (Stream.of(pipelineId, description, modelId, type, inputFieldName, outputFieldName).allMatch(x -> x != null)) {
+                try {
+                    configuration = BytesReference.bytes(
+                        buildIngestPipelineRequestContent(description, modelId, type, inputFieldName, outputFieldName)
+                    );
+                } catch (IOException e) {
+                    logger.error("Failed to create ingest pipeline configuration: " + e.getMessage());
+                    createIngestPipelineFuture.completeExceptionally(e);
+                }
+                break;
+            }
+        }
+
+        if (configuration == null) {
+            // Required workflow data not found
+            createIngestPipelineFuture.completeExceptionally(new Exception("Failed to create ingest pipeline, required inputs not found"));
+        } else {
+            // Create PutPipelineRequest and execute
+            PutPipelineRequest putPipelineRequest = new PutPipelineRequest(pipelineId, configuration, XContentType.JSON);
+            clusterAdminClient.putPipeline(putPipelineRequest, ActionListener.wrap(response -> {
+                logger.info("Created ingest pipeline : " + putPipelineRequest.getId());
+
+                // PutPipelineRequest returns only an AcknowledgeResponse, returning pipelineId instead
+                createIngestPipelineFuture.complete(new WorkflowData() {
+                    @Override
+                    public Map<String, Object> getContent() {
+                        return Map.of("pipelineId", putPipelineRequest.getId());
+                    }
+                });
+
+                // TODO : Use node client to index response data to global context (pending global context index implementation)
+
+            }, exception -> {
+                logger.error("Failed to create ignest pipeline : " + exception.getMessage());
+                createIngestPipelineFuture.completeExceptionally(exception);
+            }));
+        }
+
+        return createIngestPipelineFuture;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    /**
+     * Temporary, generates the ingest pipeline request content for text_embedding processor from workflow data
+     * {
+     *  "description" : "<description>",
+     *  "processors" : [
+     *      {
+     *          "<type>" : {
+     *              "model_id" : "<model_id>",
+     *              "field_map" : {
+     *                  "<input_field_name>" : "<output_field_name>"
+     *          }
+     *      }
+     *  ]
```

Comment:
I only did one level of nested JSON objects in my template parsing in #47 for "generic" requests assuming the "processor" field is a string-to-string map.  I could modify it to read arbitrarily nested string-to-object maps, but I'm not sure that's a good idea. 

Is this processor format consistent enough that we can generalize this "temporary" parsing when reading in a template and always do it for a "processors" key?

Reply:
All pipeline configuration formats follow a similar pattern :
```
{
    "description" : "",
    "processors": [
        {
             "processor1" : {
                 // required inputs
             }
         },
         {
             "processor2" : {
                 // required inputs
             }
         },
         ....
         ...
    ]
}
```

We can definitely generalize the creation of the configuration as long as the workflow data has a list of processors and their required inputs. The processors themselves vary in terms of format, so doing it this way would require some xcontent parser for each type of processor, since we'll need to write to specific fields.